src/master-playlist-controller.js

```diff
@@ -1325,38 +1339,44 @@ export class MasterPlaylistController extends videojs.EventTarget {
    * indefinitely.
    * @private
    */
-  excludeIncompatibleVariants_(media) {
-    let codecCount = 2;
-    let videoCodec = null;
-    let codecs;
-
-    if (media.attributes.CODECS) {
-      codecs = parseCodecs(media.attributes.CODECS);
-      videoCodec = codecs.videoCodec;
-      codecCount = codecs.codecCount;
-    }
+  excludeIncompatibleVariants_(codecString) {
+    const codecs = parseCodecs(codecString);
+    const codecCount = Object.keys(codecs).length;
 
-    this.master().playlists.forEach(function(variant) {
-      let variantCodecs = {
-        codecCount: 2,
-        videoCodec: null
-      };
+    this.master().playlists.forEach((variant) => {
+      // skip variants that are already blacklisted forever
+      if (variant.excludeUntil === Infinity) {
+        return;
+      }
+      // TODO: should we really assume codecs?
```

Comment:
Is this asking if we should assume that there are two codecs in a playlist without codec info?

Reply:
yeah

Reply:
We may want to be careful in those cases. I'm skeptical of any manifests that present wrong information or don't follow the spec. We can support them, but the support will come at a cost of possible wasted bandwidth for playlists that follow the spec.

Reply:
maybe, the ones I've seen have generally worked fine once we added the audio aac codec into the manifest.